service/src/main/kotlin/no/nav/su/se/bakover/service/AccessCheckProxy.kt

```diff
@@ -0,0 +1,216 @@
+                }
+            },
+            behandling = object : BehandlingService {
+                override fun hentBehandling(behandlingId: UUID): Either<FantIkkeBehandling, Behandling> {
+                    return services.behandling.hentBehandling((behandlingId))
+                }
+
+                override fun underkjenn(
+                    begrunnelse: String,
+                    attestant: NavIdentBruker.Attestant,
+                    behandling: Behandling
+                ): Either<Behandling.KunneIkkeUnderkjenne, Behandling> {
+                    return services.behandling.underkjenn(begrunnelse, attestant, behandling)
+                }
+
+                override fun oppdaterBehandlingsinformasjon(
+                    behandlingId: UUID,
+                    behandlingsinformasjon: Behandlingsinformasjon
+                ): Behandling {
+                    return services.behandling.oppdaterBehandlingsinformasjon(behandlingId, behandlingsinformasjon)
+                }
+
+                override fun opprettBeregning(
+                    behandlingId: UUID,
+                    fraOgMed: LocalDate,
+                    tilOgMed: LocalDate,
+                    fradrag: List<Fradrag>
+                ): Behandling {
+                    return services.behandling.opprettBeregning(behandlingId, fraOgMed, tilOgMed, fradrag)
+                }
+
+                override fun simuler(
+                    behandlingId: UUID,
+                    saksbehandler: NavIdentBruker
+                ): Either<SimuleringFeilet, Behandling> {
+                    return services.behandling.simuler(behandlingId, saksbehandler)
+                }
+
+                override fun sendTilAttestering(
+                    sakId: UUID,
+                    behandlingId: UUID,
+                    saksbehandler: NavIdentBruker.Saksbehandler
+                ): Either<KunneIkkeSendeTilAttestering, Behandling> {
+                    return services.behandling.sendTilAttestering(sakId, behandlingId, saksbehandler)
+                }
+
+                override fun iverksett(
+                    behandlingId: UUID,
+                    attestant: NavIdentBruker.Attestant
+                ): Either<Behandling.IverksettFeil, Behandling> {
+                    return services.behandling.iverksett(behandlingId, attestant)
+                }
+
+                override fun opprettSøknadsbehandling(
+                    sakId: UUID,
+                    søknadId: UUID
+                ): Either<KunneIkkeOppretteSøknadsbehandling, Behandling> {
+                    return services.behandling.opprettSøknadsbehandling(sakId, søknadId)
+                }
+            },
+            sak = object : SakService {
+                override fun hentSak(sakId: UUID): Either<FantIkkeSak, Sak> {
+                    return services.sak.hentSak(sakId)
+                        .map {
+                            assertHarTilgangTilPerson(it.fnr)
```

Comment:
På de 2 andre gjør du assertHarTilgangTilPerson før du kaller service

Reply:
Ja. Grunnen til at jeg la det slik er for at servicen heller skal få returnere `FantIkkeSak` i stedet for at det kastes en exception om at "bruker ikke har tilgang"/"noe random gikk galt". Litt usikker på hva som blir den beste måten å gjøre disse på. Det slår ekstra hardt inn på service-kall som _gjør noe_; f.eks. oppdatering av behandlingsinformasjon. Der kan vi jo ikke gjøre greiene først for så å sjekke tilgang etterpå.

Jeg heller kanskje mot at `assert`-funksjonene bare ikke gjør noenting dersom det f.eks. ikke finnes noe fødselsnummer for en gitt behandlingsId. Det er jo på en måte uansett servicene sitt ansvar å sjekke.

Reply:
Ja, jeg skjønner at det er litt tosidig og at vi ikke får en ordentlig ansvarsseparasjon her. Men det hadde vært fint med en enhetlig løsning, siden vi uansett får problematikken for mange kall.

Jeg tenker kanskje et kompromiss er at dersom det er en typisk NotFound eller BadRequest (tror ikke vi kommer borti den), så kan proxyen returnere Either.left, istedet for å kaste? Hvis ikke måtte vi kanskje håndtert forskjellige status-koder basert på type exception i den exception-handlern?